core/src/main/java/com/flipkart/poseidon/legoset/PoseidonLegoSet.java

```diff
@@ -173,4 +258,36 @@ public void setDataSourceExecutor(ExecutorService dataSourceExecutor) {
     public ExecutorService getDataSourceExecutor() {
         return dataSourceExecutor;
     }
+
+    public void setContext(ApplicationContext context) {
+        this.context = context;
+    }
+
+    private void resolveInjectableConstructorDependencies(Constructor<?> constructor, Object[] initParams, int offset, Request request) throws MissingInformationException, ElementNotFoundException {
+        for (int i = offset; i < constructor.getParameterCount(); i++) {
+            final RequestAttribute requestAttribute = constructor.getParameters()[i].getAnnotation(RequestAttribute.class);
+            final com.flipkart.poseidon.datasources.ServiceClient serviceClientAttribute = constructor.getParameters()[i].getAnnotation(com.flipkart.poseidon.datasources.ServiceClient.class);
+            if (requestAttribute != null) {
+                final String attributeName = StringUtils.isNullOrEmpty(requestAttribute.value()) ? constructor.getParameters()[i].getName() : requestAttribute.value();
+                initParams[i] = request.getAttribute(attributeName);
+            } else if (serviceClientAttribute != null) {
+                ServiceClient serviceClient = serviceClientsByName.get(constructor.getParameterTypes()[i].getName());
+                if (serviceClient == null) {
+                    throw new ElementNotFoundException("Unable to find ServiceClient for class = " + constructor.getParameterTypes()[i]);
+                }
+                initParams[i] = serviceClient;
+            } else {
+                final Qualifier qualifier = constructor.getParameters()[i].getAnnotation(Qualifier.class);
+                String beanName = null;
+                if (qualifier != null && !StringUtils.isNullOrEmpty(qualifier.value())) {
+                    beanName = qualifier.value();
+                }
+
+                initParams[i] = beanName == null ? context.getBean(constructor.getParameterTypes()[i]) : context.getBean(beanName, constructor.getParameterTypes()[i]);
```

Comment:
Good that we can inject spring beans into datasource and filter, but I don't know how performant it will be given that we construct datasource in each request at runtime. Is there a real use case for it?

Reply:
did a small test, the max time taken was for the first call which was ~1ms, rest were in low nanoseconds